File: In_class_Exercise_1/Car.cpp
#include <iostream>
#include <string>
using namespace std;

class Car {
    private:
        string brand;
        string model;
        int year;

    public:
        Car() : brand("Unknown"), model("Unknown"), year(2026) {};
        Car(string b, string m, int y) : brand(b), model(m), year(y) {}

        string getBrand() {return this->brand;}
        string getModel() {return this->model;}
        int getYear() {return this->year;}

        void setBrand(string b) {this->brand = b;}
        void setModel(string m) {this->model = m;}
        void setYear(int y) {this->year = y;}
};

int main()  {
    // Create an object using the default constructor and display its details.
    Car car1;
    cout << "Car 1 - Brand: " << car1.getBrand() << ", Model: " << car1.getModel() << ", Year: " << car1.getYear() << endl;

    cout << "----------------------------------------" << endl;

    // Create an object using the parameterized constructor and display its details.
    Car car2("Toyota", "Camry", 2020);
    cout << "Car 2 - Brand: " << car2.getBrand() << ", Model: " << car2.getModel() << ", Year: " << car2.getYear() << endl;
}

File: In_class_Exercise_1/Car.java
package In_class_Exercise_1;

public class Car {
    private String brand;
    private String model;
    private int year;

    public Car() {
        this.brand = "Unknown";
        this.model = "Unknown";
        this.year = 2026;
    }

    public Car(String brand, String model, int year) {
        this.brand = brand;
        this.model = model;
        this.year = year;
    }

    public String getBrand() { return brand; }

    public String getModel() { return model; }

    public int getYear() { return year; }

    public void setBrand(String b) { this.brand = b; }

    public void setModel(String m) { this.model = m; }

    public void setYear(int y) { this.year = y; }

    public static void main(String[] args) {
        // Create an object using the default constructor and display its details.
        Car car1 = new Car();
        System.out.println("Brand: " + car1.getBrand() + ", Model: " + car1.getModel() + ", Year: " + car1.getYear());

        // Create object using parameterized constructor
        Car car2 = new Car("Toyota", "Camry", 2024);
        System.out.println("Brand: " + car2.getBrand() + ", Model: " + car2.getModel() + ", Year: " + car2.getYear());
    }
}
